swarm/network/kademlia/kademlia.go

```diff
@@ -24,6 +24,13 @@ import (
 	"time"
 
 	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/metrics"
+)
+
+//metrics variables
+var (
+	bucketOnIndexGauge  = metrics.NewGauge("network.kademlia.bucket.onindex")
```

Comment:
`network.kademlia.bucket.onindex` isn't very descriptive, how about `network.kademlia.table.add`?

Reply:
Also, is it worth having a counter for the number of peers in the table (perhaps per bucket)?

Reply:
I still lack in-depth understanding of how swarm uses the kademlia table. My goal here is that the stats collected can help assessing health and identify problems with swarm networking, especially thinking of the open implementation for swarm light nodes etc. So whatever metric believed to help to this end, I am happy to add.

I understand `network.kademlia.bucket.onindex` to be the index at which a peer is being added in the kademlia table, not a general counter for peers in the kad table (which is the same as `addpeer.count`). Anyway, I believe all kademlia related metrics need to be revisited anyway when `swarm-network-rewrite` is completed.

Reply:
> I understand `network.kademlia.bucket.onindex` to be the index at which a peer is being added in the kademlia table, not a general counter for peers in the kad table

Ok so I didn't look closely enough at the usage, just the name of the metric. I thought it was a counter, not a gauge.

So in the code we are doing this for each addition to the kademlia table:

```
bucketOnIndexGauge.Update(int64(index))
```

What this is doing is setting the instantaneous value of the gauge to whatever bucket was just added to.

So say `addPeer` is called three times in a given interval, and they get added to buckets 3, 2 and 1 (in that order), then the gauge will have a value of 1. But what does that tell us? I don't think it is a useful metric this way. I think the gauge should be something like `network.kademlia.table.count` which represents the instantaneous count of peers in the kademlia table.

What I also may want to know in this situation is that, in this interval, a peer got added to buckets 1, 2, and 3, so I'd like to see 3 gauges get set which show the addition of these peers to those buckets (e.g. `network.kademlia.bucket.{n}.count`).

WDYT?